### PR TITLE
feat: add test build Fabric workflows

### PR DESCRIPTION
**Pull request by kacperkapusciak**
## Description

This PR adds two new workflow files - `android-build-test-fabric.yml` & `ios-build-test-fabric.yml `. These workflows test whether `react-native-screens` build without errors using Fabric architecture on both Android and iOS.

## Checklist

- [ ] Ensured that CI passes

### CODE REVIEW

File: .github/workflows/android-build-test-fabric.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Test Android build on Fabric
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'android/**'
```
Comment (Ubax) [inline .github/workflows/android-build-test-fabric.yml:7]:
```suggestion
      - 'android/**'
      - 'common/**'
      - 'src/fabric/**'
      - 'FabricExample/**'
      - 'package.json'
      - 'RNScreens.podspec'
```

File: .github/workflows/ios-build-test-fabric.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: Test iOS build on Fabric
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'ios/**'
```
Comment (Ubax) [inline .github/workflows/ios-build-test-fabric.yml:7]:
```suggestion
      - 'ios/**'
      - 'common/**'
      - 'src/fabric/**'
      - 'FabricExample/**'
      - 'package.json'
      - 'RNScreens.podspec'
```

File: .github/workflows/android-build-test-fabric.yml (hunk 1 above)
Comment (Ubax) [reply .github/workflows/android-build-test-fabric.yml:7]: We would need to add `package.json` because it can affect codegen. 
What do you think about also adding `FabricExample` here?

File: .github/workflows/ios-build-test-fabric.yml (hunk 2 above)
Comment (Ubax) [reply .github/workflows/ios-build-test-fabric.yml:7]: We would need to add `package.json` and `RNScreens.podspec` because they can affect codegen. 
What do you think about also adding `FabricExample` here?

**Review by Ubax — COMMENTED** [review]
I added some comments. I think that this is a very good idea to test fabric. We just need to remember that it is affected by much more files then paper.

**Comment by kacperkapusciak** [comment]
Thanks @Ubax! :tada: :shipit:

**Review by Ubax — APPROVED** [review]
Looks good to me 🚀

File: .github/workflows/android-build-test-fabric.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      - 'common/**'
+      - 'src/fabric/**'
+      - 'FabricExample/**'
+      - 'package.json'
+      - 'RNScreens.podspec'
```
Comment (WoLewicki) [inline .github/workflows/android-build-test-fabric.yml:12]: Could you tell why is this line needed?

**Review by WoLewicki — APPROVED** [review]
Please answer my comment, other than that LGTM!

File: .github/workflows/android-build-test-fabric.yml (hunk 3 above)
Comment (Ubax) [reply .github/workflows/android-build-test-fabric.yml:12]: You are right
There is no need for podspec on Android
It is just copy paste mistake

Comment (kacperkapusciak) [reply .github/workflows/android-build-test-fabric.yml]: Fixed in [2bbc895](https://github.com/software-mansion/react-native-screens/pull/1321/commits/2bbc8959824ed82c836501608ef66df7140e7e7e)
